### PR TITLE
Elasticsearch autocomplete

### PR DESCRIPTION
**Pull request by martaswiatkowska**
Added elasticsearch autocomplete. 
I use https://github.com/afcapel/stimulus-autocomplete but copied to our repository.

### CODE REVIEW

File: app/javascript/controllers/autocomplete_controller.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    const params = new URLSearchParams(url.search.slice(1))
+    params.append('q', query)
+    url.search = params.toString()
+
+    this.element.dispatchEvent(new CustomEvent('loadstart'))
+
+    fetch(url.toString())
+      .then(response => response.text())
+      .then(html => {
+        console.log(this.data)
```
Comment (bwilk) [inline app/javascript/controllers/autocomplete_controller.js]: remove logs

File: app/controllers/services_controller.rb [hunk 2]
```diff
@@ -25,6 +25,26 @@ def show
...
+      fields: ["title"],
+      operator: "or",
+      match: :word_middle,
+      limit: 5,
+      load: false,
+      highlight: { tag: "<b>" }
+    )
+    html_results = ""
+    query.highlights.each_with_index do |s, i|
+      html_results += "<li class=\"dropdown-item\" role=\"option\" data-autocomplete-value=\"#{i}\">#{s[:title]}</li>"
```
Comment (bwilk) [inline app/controllers/services_controller.rb]: This should be changed to 

~~~~ 
query.with_highlights.each do |s, h|
  html_results += "<li class=\"dropdown-item\" role=\"option\" data-autocomplete-value=\"#{s.title}\">#{h[:title]}</li>"
end
~~~~

**Comment by agpul** [comment]
Now it works correctly but I have additional comment. The mouse cursor is changing to something strange when I want to choose the service. It makes me feel that I can't click. Please have a look.
kursor

**Comment by martaswiatkowska** [comment]
@agpul  please check it again in other browser. I don't know what's happened because  I can't reproduce the error.

File: app/controllers/services_controller.rb (hunk 2 above)
Comment (mkasztelnik) [reply app/controllers/services_controller.rb]: More compact version:

```ruby
query.highlights.map.with_index { |s, i| ... }.join
# of if you don't need index
query.highlights.map { |s| ... }.join
```

**Comment by mkasztelnik** [comment]
Also please add changelog entry

**Pull request merged**